tests/acceptance/base-nested-engine-test.js

```diff
@@ -0,0 +1,24 @@
+import { module, test } from 'qunit';
+import { visit, currentURL } from '@ember/test-helpers';
+import { setupApplicationTest } from 'ember-qunit';
+
+module('Acceptance | base + nested engines', function(hooks) {
+  setupApplicationTest(hooks);
+
+  test('visiting /base-engine', async function(assert) {
```

Comment:
you should write tests to lazyLoading enabled too.

Reply:
@thoov can you also add unit tests in /node-tests/ultils/engine-addon-test.js ? to make sure it will work fine in the Fastboot environment.